tests/kani/DynTrait/drop_nested_boxed_dyn.rs

```diff
@@ -5,6 +5,12 @@
 // There is an implicit self-recursive call to drop_in_place, so we
 // need to set an unwind bound.
 
+// Temporarily disabling assertion reachability checks because they trigger a
+// crash in CBMC:
+// https://github.com/diffblue/cbmc/issues/6691
+// https://github.com/model-checking/kani/issues/861
+// kani-flags: --no-assertion-reach-checks
```

Comment:
This is a lot of these comment blocks. It looks like by maybe Monday the new github macos image will be deployed (usually it is by now, but apparently it's only at 3% of fleet this week), and that would let us upgrade to 5.52 which supposedly has the fix.

Planning on doing that version bump and fix all these early next week?

Reply:
> Planning on doing that version bump and fix all these early next week?

I have the branch for bumping up the version ready in https://github.com/zhassan-aws/kani/commit/ea2026570beb546afc4b1e447f184da214128a24. I was hoping that 5.52.0 would be available in github today to avoid needing to turn reachability off for those tests. Would you prefer that I hold this PR off until we get the new version of CBMC?